rbc/external.py

```diff
@@ -44,27 +47,28 @@ def __init__(self, name: str, signature: types.FunctionType):
 
     def register(self):
         # typing
-        from numba.core.typing.templates import (
-            make_concrete_template,
-            infer_global,
-            infer,
-        )
+        class ExternalTemplate(typing.templates.AbstractTemplate):
+            obj = self
+            key = self.name
 
-        template = make_concrete_template(
-            self.name, key=self.name, signatures=[self.signature]
-        )
-        infer(template)
-        infer_global(self, types.Function(template))
+            def generic(self, args, kws):
+                t = Type.fromobject(self.obj._signature).tonumba()
 
-        # lowering
-        def codegen(context, builder, sig, args):
-            fndesc = funcdesc.ExternalFunctionDescriptor(
-                self.name, sig.return_type, sig.args
-            )
-            func = context.declare_external_function(builder.module, fndesc)
-            return builder.call(func, args)
+                name = self.key
```

Comment:
When external is going to support multiple signatures, as in `external(['double log2(double)', 'float log2f(float)'], name='log2')`, then we should likely use:
```
name = t.name or self.key
```